src/cpp/library/exception

```diff
@@ -19,41 +23,54 @@ const char* message;
 typedef void (*terminate_handler)();
 typedef void (*unexpected_handler)();
 
-bool uncaught_exception() throw ();
+bool uncaught_exception() throw();
 
 // Default terminate function: calls abort
-void default_terminate() throw ()
+void default_terminate() throw()
 {
   __ESBMC_assert(0, "Aborted");
   // Allow now further execution.
   __ESBMC_assume(0);
 }
 
+// Default terminate function for unexpected: calls abort
+// A quick workaround model for "throw"
+void default_terminate_unexpected_exception() throw()
+{
+  __ESBMC_assert(0, "Aborted - unexpected exceptions");
+  // Allow now further execution.
+  __ESBMC_assume(0);
+}
+
 // Default unexpected function: calls terminate
-void default_unexpected() throw ()
+void default_unexpected() throw()
 {
   default_terminate();
 }
 
 terminate_handler terminate_pf=default_terminate;
 unexpected_handler unexpected_pf=default_unexpected;
 
-terminate_handler set_terminate(terminate_handler f) throw ()
+terminate_handler set_terminate(terminate_handler f) throw()
 {
   terminate_pf=f;
+  return terminate_pf;
 }
 
-unexpected_handler set_unexpected(unexpected_handler f) throw ()
+unexpected_handler set_unexpected(unexpected_handler f) throw()
 {
   unexpected_pf=f;
+  return unexpected_pf;
 }
 
 // Model for unexpected function: calls the current
 // unexpected handler function
 void unexpected()
 {
   unexpected_pf();
-  throw;
+  // TODO: For the time being, we use a terminate function as a workaround.
+  //   probably should rethrow the exception and just use throw.
+  default_terminate_unexpected_exception();
```

Comment:
`throw` would give: 
```
ERROR: Conversion of unsupported clang expr: "CXXThrowExpr" to expression
CXXThrowExpr 0x560340bc53b8 <library/exception:73:3> 'void'

ERROR: CONVERSION ERROR
```

Since our frontend currently doesn't support exception handling (which is another major feature but not necessarily needed for our verification goal), this is a quick workaround to avoid this error and deal with unexpected exception in a fairly decent way.